R/sf-methods.R

```diff
@@ -1,11 +1,22 @@
-
 # st_crs method for feature layer
 st_crs.FeatureLayer <- function(obj, ...) {
-  sf::st_crs(obj[["extent"]][["spatialReference"]][["latestWkid"]])
+  spatialReference <- obj[["extent"]][["spatialReference"]]
+
+  obj_crs <- spatialReference[["latestWkid"]] %||%
+    spatialReference[["wkid"]] %||%
+    spatialReference[["wkt"]]
```

Comment:
Yah, this is needed! Thank you. Think we can generalize this for both FeatureLayer and ImageServer? Something like

```r
arc_sr_to_crs <- function(obj) {
spatialReference <- obj[["extent"]][["spatialReference"]]

  obj_crs <- spatialReference[["latestWkid"]] %||%
    spatialReference[["wkid"]] %||%
    spatialReference[["wkt"]]

  sf::st_crs(obj_crs)
}
```

Reply:
Works for me – we'd need to replace calls some of the internal calls to st_crs since wherever it is currently using the S3 method. Maybe `arc_sr_as_crs()` to stay similar to all the `sf::st_as_` functions?

Reply:
Oh! To be even clearer: I think we should _keep the `sf::st_crs()`_ methods. But the methods should just call the `arc_sr_as_crs()` function since the body is identical in both cases

Reply:
Would you want to call `sf::st_crs` or `arc_sr_as_crs()` internally? Assuming you'd want to keep the default crs values the same, e.g. `crs = sf::st_crs(x)`.